flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/partitioner/ForwardPartitioner.java

```diff
@@ -43,4 +44,14 @@ public int selectChannel(SerializationDelegate<StreamRecord<T>> record) {
 	public String toString() {
 		return "FORWARD";
 	}
+
+	@Override
+	public ChannelStateRescaler getUpstreamChannelStateRescaler() {
+		return ChannelStateRescaler.FIRST_CHANNEL;
+	}
+
+	@Override
+	public ChannelStateRescaler getDownstreamChannelStateRescaler() {
+		return ChannelStateRescaler.ROUND_ROBIN;
```

Comment:
Why isn't channel state distributed to the same subtask as before (or 1st if out of range)?
(ditto upstream)
Shouldn't recovered buffers sent from the upstream match what is on the downstream?

Reply:
You are right that things should match. However, for channel state there are two dimensions on how to scale: on subtask-level and on channel-level. If you think of a keyed exchange, a rescale on downstream means that the key ranges are changing. A rescale on upstream just means that we need to redistribute the channel state to an arbitrary other subtask.
That's true for all exchanges except broadcast (where we need to discard extra state). I could also opt to just have a `boolean shouldRedistributeUpstreamState` instead of using `ChannelStateRescaler`. WDYT?

Reply:
I agree about using `ROUND_ROBIN` for the upstream. 
My concern was mostly about the dowstream (which should be resolved by addressing the comment [below](https://github.com/apache/flink/pull/13735#discussion_r517176106)).

As for `boolean`, I'd prefer `ChannelStateRescaler` as it's more flexible and consistent (with downstream).